### DISCUSSION TITLE
Adjusting "this" context to specified prototype [Feature Request]

### DISCUSSION THREAD

**Discussion by loremipsumowski** [microsoft/monaco-editor#3729] [Q&A]
### Context

- [X] This issue is not a bug report. *(please use a different template for reporting a bug)*
- [X] This issue is not a duplicate of an existing issue. *(please use the [search](https://github.com/microsoft/monaco-editor/issues) to find existing issues)*

### Description

I would like to make editor for defining callbacks of events as functions' bodies (for example click on row in table). And that callback is running under context of component that was clicked (eg. table). I would like to share d.ts definition of that component to run intellisense in Monaco, but I am not sure there is possibility to change context of `this` keyword to custom prototype.

I tried to find it in issues and in Stackoverflow, but I found only one: https://stackoverflow.com/questions/48793180/setting-this-context-on-monaco-editor-for-javascript

Maybe someone has any workaround or that is possible to introduce feature about that in next releases?

**Comment by hediet** [comment]
We don't plan to support this out of the box, but there is a feature request that asks to wrap code around the editor model for type analysis. That could also solve your issue.
